src/types/common.rs

```diff
@@ -266,6 +471,24 @@ pub struct TransparentSymmetricKey {
     pub key: Vec<u8>,
 }
 
+impl TransparentSymmetricKey {
+    pub const TAG: Tag = Tag::new(0x420043);
+    pub const KEY_TAG: Tag = Tag::new(0x42003F);
+
+    pub fn fast_scan(scanner: &mut FastScanner<'_>) -> Result<Self, FastScanError> {
+        let mut scanner = scanner.scan_struct(Self::TAG)?;
+        let key = scanner.scan_bytes(Self::KEY_TAG)?.into();
+        scanner.finish()?;
```

Comment:
Why is it correct to call `.finish()` here? If this struct was being deserialized from a larger TTLV sequence the scanner will not be empty and so calling `finish()` will cause an error. What am I missing?